File: kotlin-java/src/main/java/sub1_gettersetter/GetterSetter.java
package sub1_gettersetter;

import java.time.LocalDate;

public class GetterSetter {

    /**
     * 자바에서 코틀린 클래스를 사용해본다.
     *
     * 어려운 점이 없다.
     *
     * @JvmField 만 좀 특이하다.
     */

    public static void main(String[] args) {

        Student student = new Student();

        student.name = "AAA"; // @JvmField 가 적용되어있는 프로퍼티라 코틀린 스타일로 접근 가능
        System.out.println("name : " + student.name); // @JvmField 가 적용되어있는 프로퍼티라 코틀린 스타일로 접근 가능

        student.setBirthDate(LocalDate.of(2023, 9, 28)); // 가변 변수는 setter, getter 로 접근이 가능하다.
        System.out.println("birthDate : " + student.getBirthDate()); // 가변 변수는 setter, getter 로 접근이 가능하다.

//        student.setAge(20); // age 는 불변 변수이므로 setter 가 없다.
        System.out.println("age : " + student.getAge());

//        student.setGrade("BBB"); // 가변 변수이지만, setter 를 private 으로 막아놔서 접근 불가능
        student.changeGrade("CCC");
        System.out.println(student.getGrade());

    }
}

File: kotlin-java/src/main/kotlin/sub1_gettersetter/GetterSetter.kt
package sub1_gettersetter

/**
 * 코틀린에서 자바 클래스를 사용해본다.
 */
fun main() {

    val person = Person()

    // 자바 클래스 메서드 선언 그대로 사용할 수 도 있지만..
    person.getAge()
    person.setAge(10)

    // getter, setter 가 모두 선언 되어 있을 경우..
    // 코틀린 스타일로 사용가능하다.
    person.age = 10
    println(person.age)

    // Person 에서는 UUID 라는 프로퍼티는 없으나 getUUID 라는 메서드가 있다.
    // 마치 프로퍼티가 선언되어있는것 처럼 사용할 수 있다. (물론 getter 만 동작)
    println(person.uuid)

    // Person 에서는 myAddress 메서드로 getter 를 대신하고 있고 setAddress 는 존재한다..
    // 코틀린 스타일로 setter getter 사용불가
    // -> 즉, 코틀린 스타일로 사용하기 위해서는 getter 가 정상적으로 되어있어야 함을 알 수 있다.
//    person.address = "seoul" // compile error
//    println(person.address) // compile error
    person.setAddress("seoul")
    println(person.myAddress())
}